### PR TITLE
Mitschriften Systementwurf 31.05.2021

### PR DESCRIPTION
**Pull request by RvNovae**
~~Wir reden nicht über den linter~~

### CODE REVIEW

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWE-KASCHE/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ABWL-LEDER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/appendix/praktikum.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/DBS-DORENDORF/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/PRO-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWS-KUSCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/EMB-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/OOP-CASSELT/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- SWE-KASCHE/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

File: SWE-KASCHE/README.md [hunk 1]
```diff
@@ -695,5 +697,213 @@ public:
...
+class WrapperBibliothek { // verwendet Bibliothek2 um intern den Radius abzufragen
+  getR();
+};
+// -> Schnittstelle ist korrekt, wir verwenden aber die "schöne" Klasse/Funktion
+```
+
+**Bridge**
+
+- Implementierungen und Schnittstellen sollen getrennt werden
+- $\rightarrow$ beide voneinander unabhängig änderbar
```
Comment (ZeroPointMax) [inline SWE-KASCHE/README.md:731]:
```suggestion
- Implementierungen und Schnittstellen sollen getrennt werden $\rightarrow$ beide voneinander unabhängig änderbar
```

File: SWE-KASCHE/README.md [hunk 2]
```diff
@@ -695,5 +697,213 @@ public:
...
+
+- Einsatz bei Baumstrukturen
+- Unterschied zwischen einzelnen und zusammengesetzten Objekten soll verborgen werden
+- bspw. Baum $\rightarrow$ ein Element ohne Kinder wird anders dargestellt, als ein Element mit Kindern
+- bspw. Dateisystem $\rightarrow$
+  - Ordner (bestehen entweder selbst wieder aus Ordnern oder aus Files)
+  - File
+- bspw. Hierarchien von UI-Elementen
+  - Grafik-Elemente
+    - Elypse, Kreis, Rechteck
```
Comment (ZeroPointMax) [inline SWE-KASCHE/README.md:770]:
```suggestion
    - Ellipse, Kreis, Rechteck
```

File: SWE-KASCHE/README.md [hunk 3]
```diff
@@ -695,5 +697,213 @@ public:
...
+
+class NonTerminatingAusdruck : Ausdruck {
+  Interpretiere(k);
+};
+
+// alle abgeleiteten Klassen implementieren das "Interpretiere(k)"
+```
+
+- ein Syntaxbaum aus terminierenden und nicht-terminierenden Ausdrücken wird aufgebaut
+- Klient wertet den Syntaxbaum aus, indem er `interpretire(kontext)` aufruft
```
Comment (ZeroPointMax) [inline SWE-KASCHE/README.md:836]:
```suggestion
- Klient wertet den Syntaxbaum aus, indem er `interpretiere(kontext)` aufruft
```

File: SWE-KASCHE/README.md [hunk 4]
```diff
@@ -695,5 +697,213 @@ public:
...
+};
+```
+
+Warum Iteratoren?
+
+- es kann auf ein Element zugegriffen werden ohne, dass die interne Struktur offengelegt werden muss
+- es kann auf verschiedene Arten traversiert werden
+- es kann auf verschiedene Arten traversiert werden, ohne die Datencontainerklasse ändern zu müssen
+- es kann über eine einheitliche Schnittstelle traversiert werden
+- einer Datencontainerklasse können mehrere Iteratoren zugeordnet werden
```
Comment (ZeroPointMax) [inline SWE-KASCHE/README.md:869]: Wunderschön deutsches Wort :D

File: SWE-KASCHE/README.md [hunk 5]
```diff
@@ -695,5 +697,213 @@ public:
...
+**Qualitätsverbesserung**
+
+- Kompetenz der Entwickler
+- Prozess (Sys-Anwendung) : MMM (Mensch, Maschine, Methode)
+- Tooling, Wiederverwendung (System-Entwurf)
+- Automatisierung
+  1. DevOps Ideen
+  2. Dokumentations-Generatoren
+  3. Code-Generatoren $\rightarrow$ Galerenarbeit (generiert aus abstrakter Beschreibung)
+  4. Template-Funktionen ($\rightarrow$ `Stack<Kogel>` (ein Stack vom Typ Kogel))
```
Comment (ZeroPointMax) [inline SWE-KASCHE/README.md:880]: hehe

**Review by ZeroPointMax — APPROVED** [review]
👏

File: SWE-KASCHE/README.md (hunk 2 above)
Comment (RvNovae) [reply SWE-KASCHE/README.md:771]: OHA 😂 😂 😂

File: SWE-KASCHE/README.md (hunk 4 above)
Comment (RvNovae) [reply SWE-KASCHE/README.md:870]: Danke danke :D

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWE-KASCHE/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ABWL-LEDER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/ASM-GUENTHER/appendix/praktikum.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/DBS-DORENDORF/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/PRO-KASCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/SWS-KUSCHE/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/EMB-GUENTHER/README.md
  - /home/runner/work/dhge-pi19-sem4/dhge-pi19-sem4/OOP-CASSELT/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- SWE-KASCHE/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by ZeroPointMax** [comment]
Hm OK, das war sinnlos; die einzigen Änderungen stammen von VSCode

**Review by nukerxy — APPROVED** [review]

**Pull request merged**